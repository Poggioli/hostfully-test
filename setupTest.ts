import { vi } from "vitest";
import { QueryClient, QueryClientProvider } from "react-query";

vi.mock('axios', () => {
  return {
    default: {
      post: vi.fn(),
      get: vi.fn(),
      delete: vi.fn(),
      put: vi.fn(),
      create: vi.fn().mockReturnThis(),
      interceptors: {
        request: {
          use: vi.fn(),
          eject: vi.fn(),
        },
        response: {
          use: vi.fn(),
          eject: vi.fn(),
        },
      },
    },
  };
});

export const queryClientMock = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false
    }
  }
})
