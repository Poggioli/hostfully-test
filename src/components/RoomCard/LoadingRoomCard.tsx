import {
  Card
} from "../ui/card";
import { FC } from "react";
import { Skeleton } from "../ui/skeleton";


const LoadingRoomCard: FC = () => {
  return (
    <Card className="w-full">
      <Skeleton className="h-[200px] w-full" />
      <div className="p-3 flex flex-col gap-4">
        <Skeleton className="h-6 w-full max-w-[150px]" />
        <Skeleton className="h-14 w-full" />
        <Skeleton className="h-5 w-full max-w-[100px]" />

        <div className="flex justify-between gap-4">
          <Skeleton className="h-9 w-full max-w-[100px]" />
          <Skeleton className="h-9 w-full max-w-[100px]" />
        </div>
      </div>
    </Card>
  )
}

export { LoadingRoomCard };
