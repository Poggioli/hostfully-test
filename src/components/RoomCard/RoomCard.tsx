import { Room } from "@/service/Room"
import { FC } from "react"
import { Card, CardContent, CardDescription, CardFooter, CardHeader, CardTitle } from "../ui/card"
import { Button } from "../ui/button"
import { AspectRatio } from "@radix-ui/react-aspect-ratio"

export type RoomCardProps = Room

const RoomCard: FC<RoomCardProps> = ({ name, description, pricePerDay, photos }) => {

  return (
    <Card className="w-full">
      <AspectRatio ratio={16 / 9} className="rounded overflow-hidden">
        <img
          src={photos[0]}
          alt={`Photo of ${name}`}
          className="rounded object-cover"
        />
      </AspectRatio>
      <CardHeader className="flex flex-col gap-4 pb-3">
        <CardTitle>{name}</CardTitle>
        <CardDescription className="line-clamp-3">{description}</CardDescription>
      </CardHeader>

      <CardContent className="flex justify-end pb-4">
        <span>
          <span className="text-sm font-bold">
            ${pricePerDay}
          </span>
          {" "}night
        </span>
      </CardContent>

      <CardFooter className="flex justify-end">
        <Button>Schedule</Button>
      </CardFooter>
    </Card>
  )
}

export { RoomCard }
