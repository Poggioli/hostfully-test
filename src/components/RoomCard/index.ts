export * from './LoadingRoomCard'
export * from './RoomCard'
