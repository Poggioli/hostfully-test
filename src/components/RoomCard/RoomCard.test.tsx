import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import { RoomCard } from './RoomCard'
import { BrowserRouter } from 'react-router-dom';
import { QueryClientProvider } from 'react-query';
import { queryClientMock } from '../../../setupTest';

describe('RoomCard', () => {
  it(`GIVEN a RoomCard
      WHEN render
      THEN should match with snapshot`, () => {
    const room = {
      id: "82c67959-a414-4470-b26b-ba4261cfe867",
      name: "House with Wi-Fi",
      description: "Is located just 1 hour from the center of SP, easy access via Regis Bitencourt, an all paved route",
      pricePerDay: 900,
      photos: [
        "https://images.unsplash.com/photo-1704211825599-9e2ec712f561",
        "https://images.unsplash.com/photo-1696861080288-0cc2f1cd48d5"
      ]
    }
    const { baseElement } = render(
      <BrowserRouter>
        <QueryClientProvider client={queryClientMock}>
          <RoomCard {...room} />
        </QueryClientProvider>
      </BrowserRouter>
    )
    expect(baseElement).toMatchSnapshot()
  });
})