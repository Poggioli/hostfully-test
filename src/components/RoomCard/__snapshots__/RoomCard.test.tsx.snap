// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RoomCard > GIVEN a RoomCard
      WHEN render
      THEN should match with snapshot 1`] = `
<body>
  <div>
    <div
      class="rounded-xl border bg-card text-card-foreground shadow w-full"
    >
      <div
        data-radix-aspect-ratio-wrapper=""
        style="position: relative; width: 100%; padding-bottom: 56.25%;"
      >
        <div
          class="rounded overflow-hidden"
          style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
        >
          <img
            alt="Photo of House with Wi-Fi"
            class="rounded object-cover"
            src="https://images.unsplash.com/photo-1704211825599-9e2ec712f561"
          />
        </div>
      </div>
      <div
        class="space-y-1.5 p-6 flex flex-col gap-4 pb-3"
      >
        <h3
          class="font-semibold leading-none tracking-tight"
        >
          House with Wi-Fi
        </h3>
        <p
          class="text-sm text-muted-foreground line-clamp-3"
        >
          Is located just 1 hour from the center of SP, easy access via Regis Bitencourt, an all paved route
        </p>
      </div>
      <div
        class="p-6 pt-0 flex justify-end pb-4"
      >
        <span>
          <span
            class="text-sm font-bold"
          >
            $
            900
          </span>
           
          night
        </span>
      </div>
      <div
        class="items-center p-6 pt-0 flex justify-end"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
        >
          Schedule
        </button>
      </div>
    </div>
  </div>
</body>
`;
